### DISCUSSION TITLE
Undo/Redo Buttons

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#5235] [Ideas]
Undoing/redoing a list of steps is really annoying if you have to right-click the script area at each step.

Adding undo/redo buttons to the script area that replicate the behaviour of the undo/redo options in the right-click menu would be a very useful shortcut.

**Comment by AstralButOnGitHub** [comment]
I like it!
But what would it look like?

**Reply by sfederici** [reply]
> what would it look like?

I saw them in similar tools, they are a back (e.g. https://www.flaticon.com/free-icon/turn-left_7030378) and a forward arrow (e.g. https://www.flaticon.com/free-icon/right-arrow_7903573) at the left-hand side of the semitransparent sprite icon in the code area.

**Reply by AstralButOnGitHub** [reply]
You're suggesting something like this?
![image](https://user-images.githubusercontent.com/116039257/197279539-5db94e28-c4f3-4c33-899a-1c4b3060e4be.png)

**Reply by sfederici** [reply]
Yes. Or maybe putting the arrows above the cat icon (or at the left).

**Reply by AstralButOnGitHub** [reply]
![image](https://user-images.githubusercontent.com/116039257/197284663-ddfee208-03c5-4219-8315-dd576d3c03df.png)
Like this?

**Reply by BroJac5246** [reply]
> ![image](https://user-images.githubusercontent.com/116039257/197284663-ddfee208-03c5-4219-8315-dd576d3c03df.png)
> Like this?

It would look better above, though

**Reply by sfederici** [reply]
Yes, one of the alternatives.

**Reply by sfederici** [reply]
> It would look better above, though

I guess so
